### PR TITLE
Code clean-up and enhancements, storing data on disk, loading data from a file, output normalization

### PR DESCRIPTION
**Pull request by cgrambow**
## Motivation

The goal is to make this package stand-alone and suitable for any type of data. To create a more official package, the first step is to clean up the code, i.e., enhance PEP 8 compliance. Next, several features are added to improve versatility of the package.

## Code clean-up (better PEP 8 compliance)

- Change tabs to spaces
- Stylistic changes (no relative imports, remove unused code, no `*` imports, fix unusual indents and spaces, etc.)

## Enhancements

- Allow `list` and `numpy.ndarray` data
- More efficient array shuffling
- Save memory by not copying large arrays
- Unit tests to test neural net training

## New features

- Enable storing `molecule_tensor`s on disk instead of in memory to greatly reduce memory requirements
- Enable loading data from file (`.csv` format; first column is list of SMILES, remaining columns are output data)
- Add options for specifying training and testing ratios
- Save model after each epoch
- Add option to load pretrained weights for transfer learning
- Add option to normalize output data and save mean and standard deviation to file

## Future changes

This PR is big enough as is, so I'm just listing some future changes I plan to make.
- Remove RMG dependency and replace with RDKit (maybe Open Babel)
- Refactor code, most likely renaming module (`cnn_framework` made sense when this was a part of RMG, but not anymore)
- Add improved pretrained models
- Further improve code versatility by not tying it to specific types of data
- Probably many other things

### CODE REVIEW

**Comment by KEHANG** [comment]
Would be a very useful PR! We can consider releasing a new version after this.

**Review by yplitw — APPROVED** [review]
